src/pages/iou/request/step/IOURequestStepScan/index.native.tsx

```diff
@@ -664,6 +670,16 @@ function IOURequestStepScan({
         [shouldSkipConfirmation, navigateToConfirmationStep, initialTransaction?.amount, iouType, shouldStartLocationPermissionFlow],
     );
 
+    const submitMultiScanReceipts = () => {
+        const transactionReceipts = (optimisticTransactionsReceipts ?? [])
+            .filter((receipt): receipt is ReceiptWithTransactionID & {source: string} => !!receipt.source)
+            .map((receipt) => ({
+                ...receipt,
+                file: receiptFiles.find((receiptFile) => receiptFile.transactionID === receipt.transactionID)?.file,
+            }));
+        submitReceipts(transactionReceipts);
+    };
```

Comment:
In this case, we already have existing optimisticTransactions, so I don’t think we need to use `getTransactionDraftReceiptsSelector`


```suggestion
    const submitMultiScanReceipts = () => {
        const transactionIDs = new Set(
            optimisticTransactions?.map(transaction => transaction?.transactionID)
        );
        const validReceiptFiles = receiptFiles.filter(receiptFile =>
            transactionIDs.has(receiptFile.transactionID)
        );
        submitReceipts(validReceiptFiles);
    };
```